curvefs/src/client/s3/client_s3_cache_manager.h

```diff
@@ -296,10 +314,43 @@ class FsCacheManager {
     }
 
     void DataCacheNumFetchSub(uint64_t v) {
-        assert(wDataCacheNum_.load(std::memory_order_relaxed) != 0);
+        assert(wDataCacheNum_.load(std::memory_order_relaxed) >= v);
         wDataCacheNum_.fetch_sub(v, std::memory_order_relaxed);
     }
 
+    void DataCacheByteInc(uint64_t v) {
+        LOG(INFO) << "DataCacheByteInc() v:" << v << ",wDataCacheByte:"
+                  << wDataCacheByte_.load(std::memory_order_relaxed);
+        wDataCacheByte_.fetch_add(v, std::memory_order_relaxed);
+    }
+
+    void DataCacheByteDec(uint64_t v) {
+        LOG(INFO) << "DataCacheByteDec() v:" << v << ",wDataCacheByte:"
+                  << wDataCacheByte_.load(std::memory_order_relaxed);
+        assert(wDataCacheByte_.load(std::memory_order_relaxed) >= v);
+        wDataCacheByte_.fetch_sub(v, std::memory_order_relaxed);
+    }
+
+    void WaitFlush() {
```

Comment:
It's better change this function to:
void WaitFlush() {
     std::unique_lock<std::mutex> lk(mutex_);
    isWaiting_.exchange(true, std::memory_order_acq_rel);
    cond_.wait(lk);
    }